packages/admin/admin/src/common/HoverActions.tsx

```diff
@@ -1,61 +1,98 @@
-import { ComponentsOverrides, Grow, Theme } from "@mui/material";
-import { createStyles, WithStyles, withStyles } from "@mui/styles";
+import { ComponentsOverrides, css, Grow, styled, Theme, useThemeProps } from "@mui/material";
+import { ThemedComponentBaseProps } from "helpers/ThemedComponentBaseProps";
 import * as React from "react";
 
-export type HoverActionsProps = React.PropsWithChildren<{
+export interface HoverActionsProps
+    extends ThemedComponentBaseProps<{
+        root: "div";
+        hoverAreaExpansion: "div";
+        actions: "div";
+        children: "div";
+    }> {
     actions?: React.ReactNode;
-}>;
+    children?: React.ReactNode;
+}
+
+export type HoverActionsClassKey = "root" | "hoverAreaExpansion" | "actions" | "children";
 
-const HoverActions = ({ classes, actions, children }: HoverActionsProps & WithStyles<typeof styles>): React.ReactElement => {
+export const HoverActions = (inProps: HoverActionsProps) => {
+    const { actions, children, slotProps, ...restProps } = useThemeProps({ props: inProps, name: "CometAdminHoverActions" });
     const [isHovering, setIsHovering] = React.useState<boolean>(false);
 
     return (
-        <div className={classes.root} onMouseEnter={() => setIsHovering(true)} onMouseLeave={() => setIsHovering(false)}>
-            <div className={classes.hoverAreaExpansion} />
+        <Root onMouseEnter={() => setIsHovering(true)} onMouseLeave={() => setIsHovering(false)} {...slotProps?.root} {...restProps}>
+            <HoverAreaExpansion {...slotProps?.hoverAreaExpansion} {...restProps} />
             <Grow in={Boolean(actions) && isHovering}>
-                <div className={classes.actions}>{actions}</div>
+                <Actions {...slotProps?.actions} {...restProps}>
+                    {actions}
+                </Actions>
```

Comment:
`restProps` should only be passed through to the Root slot. 

```suggestion
                <Actions {...slotProps?.actions}>
                    {actions}
                </Actions>
```